src/panels/lovelace/components/hui-energy-period-selector.ts

```diff
@@ -70,50 +76,119 @@ export class HuiEnergyPeriodSelector extends SubscribeMixin(LitElement) {
+
+    // pre defined date ranges
+    this._ranges = {
+      [this.hass.localize("ui.components.date-range-picker.ranges.today")]: [
+        calcDate(today, startOfDay, this.hass.locale, this.hass.config, {
+          weekStartsOn,
+        }),
+        calcDate(today, endOfDay, this.hass.locale, this.hass.config, {
+          weekStartsOn,
+        }),
+      ],
+      [this.hass.localize("ui.components.date-range-picker.ranges.yesterday")]:
+        [
+          calcDate(
+            addDays(today, -1),
+            startOfDay,
+            this.hass.locale,
+            this.hass.config,
+            {
+              weekStartsOn,
+            }
+          ),
+          calcDate(
+            addDays(today, -1),
+            endOfDay,
+            this.hass.locale,
+            this.hass.config,
+            {
+              weekStartsOn,
+            }
+          ),
+        ],
+      [this.hass.localize("ui.components.date-range-picker.ranges.this_week")]:
+        [
+          calcDate(today, startOfWeek, this.hass.locale, this.hass.config, {
+            weekStartsOn,
+          }),
+          calcDate(today, endOfWeek, this.hass.locale, this.hass.config, {
+            weekStartsOn,
+          }),
+        ],
+      [this.hass.localize("ui.components.date-range-picker.ranges.this_month")]:
+        [
+          calcDate(today, startOfMonth, this.hass.locale, this.hass.config),
+          calcDate(today, endOfMonth, this.hass.locale, this.hass.config),
+        ],
+      [this.hass.localize(
+        "ui.components.date-range-picker.ranges.this_quarter"
+      )]: [
+        calcDate(today, startOfQuarter, this.hass.locale, this.hass.config),
+        calcDate(today, endOfQuarter, this.hass.locale, this.hass.config),
+      ],
+      [this.hass.localize("ui.components.date-range-picker.ranges.this_year")]:
+        [
+          calcDate(today, startOfYear, this.hass.locale, this.hass.config),
+          calcDate(today, endOfYear, this.hass.locale, this.hass.config),
+        ],
+    };
+  }
+
   protected render() {
     if (!this.hass || !this._startDate) {
       return nothing;
     }
 
-    const viewButtons: ToggleButton[] = [
-      {
-        label: this.hass.localize(
-          "ui.panel.lovelace.components.energy_period_selector.day"
-        ),
-        value: "day",
-      },
-      {
-        label: this.hass.localize(
-          "ui.panel.lovelace.components.energy_period_selector.week"
-        ),
-        value: "week",
-      },
-      {
-        label: this.hass.localize(
-          "ui.panel.lovelace.components.energy_period_selector.month"
-        ),
-        value: "month",
-      },
-      {
-        label: this.hass.localize(
-          "ui.panel.lovelace.components.energy_period_selector.year"
-        ),
-        value: "year",
-      },
-    ];
-
     return html`
       <div class="row">
         <div class="label">
-          ${this._period === "day"
+          ${this._simpleRange() === "day"
```

Comment:
We should not call this function multiple times in a render, make it a const:

`const simpleRange = this._simpleRange()`
```suggestion
          ${simpleRange === "day"
```